modules/gapi/test/s11n/gapi_s11n_tests.cpp

```diff
@@ -213,6 +228,60 @@ TEST_F(S11N_Basic, Test_RunArgs_Scalar) {
     EXPECT_EQ(scalar2, out_scalar2);
 }
 
+TEST_F(S11N_Basic, Test_RunArg_Opaque) {
+    auto op = cv::detail::OpaqueRef(42);
+    auto v = cv::GRunArg{ op };
+    put(v);
```

Comment:
I'd avoid using our internal API objects directly here.

Instead, this should work:
```c++
auto args = cv::gin(42);
put(args[0]);
```

Reply:
Other tests shares the same logic. I've tried to do so, however there are some hidden problems. First, it won't determine the type during serialization if I just write
```
auto args = cv::gin(42);
put(args[0]);
```
but if I do
```
auto args = cv::gin(cv::detail::OpaqueRef(42));
put(args[0]);
```
it will only work for Opaque but not for Array. I've tried to add universal reference to `cv::gin` and it didn't help. Tried to remove `std::move` from `VectorRef` - also didn't work out. I suggest to leave it as is. However, this problem should be investigated further.

Reply:
Please file a bug

Reply:
Done